apps/vc-api/test/vc-api/exchanges/ew-role/ew-role-vc-exchange.e2e-suite.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2021, 2022 Energy Web Foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  VerifiableCredentialsServiceBase,
+  getVerifiableCredentialsService,
+  IssuerFields,
+  fromPrivateKey,
+  CacheClient,
+  RoleCredentialSubject
+} from 'iam-client-lib';
+import {
+  Credential,
+  CredentialStatusPurpose,
+  StatusListEntryType
+} from '@ew-did-registry/credentials-interface';
+import { Wallet, providers, utils } from 'ethers';
+import axios from 'axios';
+import { error } from 'console';
+// import {} from '../../../../../../common/temp/node_modules/iam-client-lib/dist/src/modules/cache-client/cache-client.service'
+
+const { id } = utils;
+
+jest.mock('axios');
+const getClaimsBySubject = jest.fn();
+jest.mock('iam-client-lib/dist/src/modules/cache-client/cache-client.service', () => {
```

Comment:
@jrhender @JGiter @jrhender Is there any way we can mock the `CacheClient` module such that we have access to intermediate results (credential argument of `addStatusToCredential` method) ? `js.mock() only takes  path of the module and `aim-client-lib` only has `Dist` build but not the `src` directory.

Reply:
I would try something like
```
const addStatusToCredential = jest.fn();
jest.mock('iam-client-lib', () => {
   return jest.fn().mockImplementation(() => {
    return {CacheClient: {addStatusToCredential}};
  });
})